### PR TITLE
Add shadow-dom support

### PR DESCRIPTION
**Pull request by dstekanov**
## Proposed changes
Add support for searching elements inside Shadow-DOM tree. Currently, it does not have native support by WebDriver so it is a good helper who faced with such elements.

In order to be able to findElement inside shadow-dom, you have to specify its shadow-host element (it is a root element that has attached shadow tree). There is a good explanation about it https://javascript.info/shadow-dom 

Also, one note is that if you have inner shadow roots, you must explicitly specify each its shadow host.

It resolves a feature request #1014 

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome_headless firefox_headless` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: statics/src/test/java/integration/ShadowDomInsideIFrameTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Test
+  void getTargetElementViaShadowHostInsideIFrame() {
+    switchTo().frame("iframe_page");
+    $(shadowCss("p", "#shadow-host"))
+      .shouldHave(text("Inside Shadow-DOM"));
+  }
+
+  @Test
+  void setValueViaShadowDomInsideIFrame() {
```
Comment (BorisOsipov) [inline statics/src/test/java/integration/ShadowDomInsideIFrameTest.java]: add assume not FF check to avoid test fail on FF

Comment (dstekanov) [reply statics/src/test/java/integration/ShadowDomInsideIFrameTest.java]: thanks! done

**Comment by Yohandah** [comment]
How to use this ?

How would you request `toast-message` in this example ?

I tried the following but it doesn't work like this

`$(By.tagName("ion-toast")).shadowRoot.$(By.className("toast-message")).getText();`


image

**Comment by QuiWorks** [comment]
From Looking at the changes in this pull request (see SelectorsTest#byShadowCss) and the [java doc](https://selenide.org/javadoc/current//com/codeborne/selenide/selector/ByShadow.html) this seems like the intended use:

`SelenideElement element = $(ByShadow.cssSelector(targetCssSelector, hostCssSelector));`

**Comment by DieterRogiest** [comment]
see example in the blog https://selenide.org/2020/03/18/selenide-5.10.0/ : 

`$(shadowCss("p", "#shadow-host", "#inner-shadow-host")).shouldHave(text("The Shadow-DOM inside another shadow tree"));`

**Pull request merged**